### DISCUSSION TITLE
Apple Certification 기간 만료에 따른 인증서 및 프로비저닝 갱신 방법 공유

### DISCUSSION THREAD

**Discussion by HanGyeongjun** [Team-ShortcutsZip/MacC-Team-HappyAnding#414] [Announcements]
## 뭔일인가?
- Certification 기간 만료로(아마도?) 실기기 빌드 시도 시 인증서가 포함이 안 되어있다는 요상한 에러가 발생합니다!

## 실기기 빌드 하려면?
1. ‘키체인 접근(Keychain Acess)’ 앱을 열고 login-Keys로 들어가주세요.
2. 경준 한 아니면 Han Gyeongjun같이 저와 관련되어있는걸로 보이는 기존의 Key를 모두 지워주세요.
3. login-Certificates도 확인해서 저와 관련되어있는거같은걸 싹 지워주세요.
4. 따로 공유해드린 .p12파일을 다운받고 더블클릭해 실행해주세요. 암호는 ********입니다. 이후 맥 로그인 암호도 입력하면 설치됩니다.
5. ‘키체인 접근’ 앱의 login-Keys에 제 이름으로 된 새로운 개인 키(private key) 가 생성되었는지 확인해주세요. 
6. 해당 개인 키 왼쪽의 드롭다운 화살표를 눌러 ‘Apple Development: Gyeongjun Han (X6…)’ 이름의 인증서가 있는지 확인해주세요. 만료일은 2024년 2월 29일입니다.
7. 공유해드린 mainApp과 Share Extension의 새로운 프로비저닝 또한 다운받아주세요.    
8. Xcode를 열어 HappyAnding 프로젝트 설정을 들어가주세요.
9. Target-HappyAnding-Signing&Capabilities의 프로비저닝 프로파일을 클릭해 새로운 프로비저닝 파일을 import해주세요.
10. Target-ShareExtension에서도 새로운 익스텐션용 프로비저닝 파일을 import해주세요.
11. 실기기에 클린빌드 후 계속해서 오류가 발생하면 알려주세요.
12. 지금 공유해드린 인증서 및 프로비저닝은 개발용이라 혹시 App Store Distribution용이 필요하다면 말씀해주세요!

**Comment by JIWON1923** [comment]
암호 이렇게 공유하셔도 되나요?

**Reply by JIWON1923** [reply]
큰일나셔요 선생님

**Reply by HanGyeongjun** [reply]
파일 안올렸으니 괜찮습니다 ㅋㅋㅋㅋㅋ

**Reply by HanGyeongjun** [reply]
블라인드 처리 하겠읍니다...ㅋㅋㅋㅋㅋㅋ

**Comment by HanGyeongjun** [comment]
인증서 발급하고 어쩌고 하는 과정도 까먹을 것 같으니까 조만간 쭉 기록해놔야겠네요
